presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -332,4 +371,24 @@ private static BytecodeBlock buildInCase(
         caseBlock.append(elseNode);
         return caseBlock;
     }
+
+    private static boolean isDeterminateConstant(RowExpression expression, MethodHandle isIndeterminateFunction)
+    {
+        if (!(expression instanceof ConstantExpression)) {
+            return false;
+        }
+        ConstantExpression constantExpression = (ConstantExpression) expression;
+        Object value = constantExpression.getValue();
+        boolean isNull = value == null;
+        if (isNull) {
+            return false;
+        }
+        try {
+            return !(boolean) isIndeterminateFunction.invoke(value, isNull);
```

Comment:
Use `InterpretedFunctionInvoker` instead

Reply:
`InterpretedFunctionInvoker` takes `ConnectorSession` as a parameter. `ConnectorSession` is not available in the compiler.